{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "cubic-paraguay",
   "metadata": {},
   "source": [
    "## Obtain exon coordinates corresponding to the 65 CH genes\n",
    "\n",
    "### CH gene list\n",
    "IntOGen-DriverGenes-CH.tsv - Obtained from www.intogen.org/ch/ (2021-02-24)\n",
    "\n",
    "### Gene annotations (Canonical transcripts)\n",
    "\n",
    "**1. Data from Biomart obtained from file from Mónica: ensembl_canonical_transcripts.tsv (2021-02-24)**\n",
    "\n",
    "ensembl_canonical_transcripts_genes.txt - Obtained from Biomart (Ensembl Genes 103 & GRCh38.p13) using previous file as Filter (2021-02-25)\n",
    "\n",
    "**2. Data from cluster (Claudia)**\n",
    "/workspace/projects/genomic_regions/hg38/release_2/cds_with_stop_codon/hg38_cds_with_stop_codon.canonical.overlap.gz\n",
    "\n",
    "Ensembl v101 & Gencode v35 coordenates\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "written-nelson",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "brutal-financing",
   "metadata": {},
   "source": [
    "### READ DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dominant-poultry",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Mutations</th>\n",
       "      <th>Samples</th>\n",
       "      <th>Samples (%)</th>\n",
       "      <th>Cohorts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>3,343</td>\n",
       "      <td>3,005</td>\n",
       "      <td>15.64</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TET2</td>\n",
       "      <td>1,011</td>\n",
       "      <td>925</td>\n",
       "      <td>4.81</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PPM1D</td>\n",
       "      <td>776</td>\n",
       "      <td>683</td>\n",
       "      <td>3.56</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASXL1</td>\n",
       "      <td>438</td>\n",
       "      <td>433</td>\n",
       "      <td>2.25</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>352</td>\n",
       "      <td>337</td>\n",
       "      <td>1.75</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>KDM5C</td>\n",
       "      <td>109</td>\n",
       "      <td>7</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>MYO5A</td>\n",
       "      <td>96</td>\n",
       "      <td>7</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>ERF</td>\n",
       "      <td>27</td>\n",
       "      <td>6</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>CUX1</td>\n",
       "      <td>76</td>\n",
       "      <td>5</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>CDKN1B</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Symbol Mutations Samples  Samples (%)  Cohorts\n",
       "0   DNMT3A     3,343   3,005        15.64        3\n",
       "1     TET2     1,011     925         4.81        3\n",
       "2    PPM1D       776     683         3.56        3\n",
       "3    ASXL1       438     433         2.25        3\n",
       "4      ATM       352     337         1.75        3\n",
       "..     ...       ...     ...          ...      ...\n",
       "60   KDM5C       109       7         0.04        1\n",
       "61   MYO5A        96       7         0.04        1\n",
       "62     ERF        27       6         0.03        1\n",
       "63    CUX1        76       5         0.03        1\n",
       "64  CDKN1B        15       4         0.02        1\n",
       "\n",
       "[65 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Obtain CH genes data\n",
    "\n",
    "ch_genes_df = pd.read_csv(\"/home/sdemajo/CH_gnomad_2021_02/data/IntOGen-DriverGenes-CH.tsv\", sep=\"\\t\")\n",
    "ch_genes_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "temporal-gentleman",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/sdemajo/miniconda3/envs/clonalhemato/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3155: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene stable ID version</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Transcript stable ID version</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Exon region start (bp)</th>\n",
       "      <th>Exon region end (bp)</th>\n",
       "      <th>Exon rank in transcript</th>\n",
       "      <th>Exon stable ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ENSG00000198888.2</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>ENST00000361390.2</td>\n",
       "      <td>MT</td>\n",
       "      <td>3307</td>\n",
       "      <td>4262</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>3307</td>\n",
       "      <td>4262</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00001435714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>ENSG00000198763.3</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>ENST00000361453.3</td>\n",
       "      <td>MT</td>\n",
       "      <td>4470</td>\n",
       "      <td>5511</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>4470</td>\n",
       "      <td>5511</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00001435686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>ENSG00000198804.2</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>ENST00000361624.2</td>\n",
       "      <td>MT</td>\n",
       "      <td>5904</td>\n",
       "      <td>7445</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>5904</td>\n",
       "      <td>7445</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00001435647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>ENSG00000198712.1</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>ENST00000361739.1</td>\n",
       "      <td>MT</td>\n",
       "      <td>7586</td>\n",
       "      <td>8269</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>7586</td>\n",
       "      <td>8269</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00001435613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>ENSG00000228253.1</td>\n",
       "      <td>ENST00000361851</td>\n",
       "      <td>ENST00000361851.1</td>\n",
       "      <td>MT</td>\n",
       "      <td>8366</td>\n",
       "      <td>8572</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>8366</td>\n",
       "      <td>8572</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00001435286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220303</th>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>ENSG00000171163.16</td>\n",
       "      <td>ENST00000451251</td>\n",
       "      <td>ENST00000451251.5</td>\n",
       "      <td>1</td>\n",
       "      <td>248850006</td>\n",
       "      <td>248859144</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>248858131</td>\n",
       "      <td>248858321</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSE00003615063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220304</th>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>ENSG00000171163.16</td>\n",
       "      <td>ENST00000451251</td>\n",
       "      <td>ENST00000451251.5</td>\n",
       "      <td>1</td>\n",
       "      <td>248850006</td>\n",
       "      <td>248859144</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>248850066</td>\n",
       "      <td>248850516</td>\n",
       "      <td>12</td>\n",
       "      <td>ENSE00001648925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220305</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>ENSG00000185220.12</td>\n",
       "      <td>ENST00000329291</td>\n",
       "      <td>ENST00000329291.6</td>\n",
       "      <td>1</td>\n",
       "      <td>248906196</td>\n",
       "      <td>248919946</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>248906235</td>\n",
       "      <td>248906342</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00001637224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220306</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>ENSG00000185220.12</td>\n",
       "      <td>ENST00000329291</td>\n",
       "      <td>ENST00000329291.6</td>\n",
       "      <td>1</td>\n",
       "      <td>248906196</td>\n",
       "      <td>248919946</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>248913816</td>\n",
       "      <td>248913879</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSE00003610993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220307</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>ENSG00000185220.12</td>\n",
       "      <td>ENST00000329291</td>\n",
       "      <td>ENST00000329291.6</td>\n",
       "      <td>1</td>\n",
       "      <td>248906196</td>\n",
       "      <td>248919946</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>248916602</td>\n",
       "      <td>248919146</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSE00001312561</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>220308 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene stable ID Gene stable ID version Transcript stable ID  \\\n",
       "0       ENSG00000198888      ENSG00000198888.2      ENST00000361390   \n",
       "1       ENSG00000198763      ENSG00000198763.3      ENST00000361453   \n",
       "2       ENSG00000198804      ENSG00000198804.2      ENST00000361624   \n",
       "3       ENSG00000198712      ENSG00000198712.1      ENST00000361739   \n",
       "4       ENSG00000228253      ENSG00000228253.1      ENST00000361851   \n",
       "...                 ...                    ...                  ...   \n",
       "220303  ENSG00000171163     ENSG00000171163.16      ENST00000451251   \n",
       "220304  ENSG00000171163     ENSG00000171163.16      ENST00000451251   \n",
       "220305  ENSG00000185220     ENSG00000185220.12      ENST00000329291   \n",
       "220306  ENSG00000185220     ENSG00000185220.12      ENST00000329291   \n",
       "220307  ENSG00000185220     ENSG00000185220.12      ENST00000329291   \n",
       "\n",
       "       Transcript stable ID version Chromosome/scaffold name  Gene start (bp)  \\\n",
       "0                 ENST00000361390.2                       MT             3307   \n",
       "1                 ENST00000361453.3                       MT             4470   \n",
       "2                 ENST00000361624.2                       MT             5904   \n",
       "3                 ENST00000361739.1                       MT             7586   \n",
       "4                 ENST00000361851.1                       MT             8366   \n",
       "...                             ...                      ...              ...   \n",
       "220303            ENST00000451251.5                        1        248850006   \n",
       "220304            ENST00000451251.5                        1        248850006   \n",
       "220305            ENST00000329291.6                        1        248906196   \n",
       "220306            ENST00000329291.6                        1        248906196   \n",
       "220307            ENST00000329291.6                        1        248906196   \n",
       "\n",
       "        Gene end (bp) Gene name  Exon region start (bp)  Exon region end (bp)  \\\n",
       "0                4262    MT-ND1                    3307                  4262   \n",
       "1                5511    MT-ND2                    4470                  5511   \n",
       "2                7445    MT-CO1                    5904                  7445   \n",
       "3                8269    MT-CO2                    7586                  8269   \n",
       "4                8572   MT-ATP8                    8366                  8572   \n",
       "...               ...       ...                     ...                   ...   \n",
       "220303      248859144    ZNF692               248858131             248858321   \n",
       "220304      248859144    ZNF692               248850066             248850516   \n",
       "220305      248919946     PGBD2               248906235             248906342   \n",
       "220306      248919946     PGBD2               248913816             248913879   \n",
       "220307      248919946     PGBD2               248916602             248919146   \n",
       "\n",
       "        Exon rank in transcript   Exon stable ID  \n",
       "0                             1  ENSE00001435714  \n",
       "1                             1  ENSE00001435686  \n",
       "2                             1  ENSE00001435647  \n",
       "3                             1  ENSE00001435613  \n",
       "4                             1  ENSE00001435286  \n",
       "...                         ...              ...  \n",
       "220303                        2  ENSE00003615063  \n",
       "220304                       12  ENSE00001648925  \n",
       "220305                        1  ENSE00001637224  \n",
       "220306                        2  ENSE00003610993  \n",
       "220307                        3  ENSE00001312561  \n",
       "\n",
       "[220308 rows x 12 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Obtain exons hg38 1 (Biomart)\n",
    "\n",
    "exons1 = pd.read_csv(\"/home/sdemajo/CH_gnomad_2021_02/data/ensembl_canonical_transcripts_exons.txt\", sep=\"\\t\")\n",
    "exons1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "framed-abortion",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/sdemajo/miniconda3/envs/clonalhemato/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3155: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>GENE_ID</th>\n",
       "      <th>TRANSCRIPT_ID</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>70008</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENST00000335137</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>450740</td>\n",
       "      <td>451678</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000284733</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>685716</td>\n",
       "      <td>686654</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>925942</td>\n",
       "      <td>926013</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000187634</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>930155</td>\n",
       "      <td>930336</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000187634</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193025</th>\n",
       "      <td>Y</td>\n",
       "      <td>25038809</td>\n",
       "      <td>25038914</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000185894</td>\n",
       "      <td>ENST00000382287</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193026</th>\n",
       "      <td>Y</td>\n",
       "      <td>25041769</td>\n",
       "      <td>25041886</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000185894</td>\n",
       "      <td>ENST00000382287</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193027</th>\n",
       "      <td>Y</td>\n",
       "      <td>25043946</td>\n",
       "      <td>25044023</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000185894</td>\n",
       "      <td>ENST00000382287</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193028</th>\n",
       "      <td>Y</td>\n",
       "      <td>25622443</td>\n",
       "      <td>25624034</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000172288</td>\n",
       "      <td>ENST00000306609</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193029</th>\n",
       "      <td>Y</td>\n",
       "      <td>25624455</td>\n",
       "      <td>25624527</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000172288</td>\n",
       "      <td>ENST00000306609</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193030 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROMOSOME     START       END STRAND          GENE_ID  \\\n",
       "0               1     69091     70008      +  ENSG00000186092   \n",
       "1               1    450740    451678      -  ENSG00000284733   \n",
       "2               1    685716    686654      -  ENSG00000284662   \n",
       "3               1    925942    926013      +  ENSG00000187634   \n",
       "4               1    930155    930336      +  ENSG00000187634   \n",
       "...           ...       ...       ...    ...              ...   \n",
       "193025          Y  25038809  25038914      -  ENSG00000185894   \n",
       "193026          Y  25041769  25041886      -  ENSG00000185894   \n",
       "193027          Y  25043946  25044023      -  ENSG00000185894   \n",
       "193028          Y  25622443  25624034      +  ENSG00000172288   \n",
       "193029          Y  25624455  25624527      +  ENSG00000172288   \n",
       "\n",
       "          TRANSCRIPT_ID  SYMBOL  \n",
       "0       ENST00000335137   OR4F5  \n",
       "1       ENST00000426406  OR4F29  \n",
       "2       ENST00000332831  OR4F16  \n",
       "3       ENST00000342066  SAMD11  \n",
       "4       ENST00000342066  SAMD11  \n",
       "...                 ...     ...  \n",
       "193025  ENST00000382287   BPY2C  \n",
       "193026  ENST00000382287   BPY2C  \n",
       "193027  ENST00000382287   BPY2C  \n",
       "193028  ENST00000306609    CDY1  \n",
       "193029  ENST00000306609    CDY1  \n",
       "\n",
       "[193030 rows x 7 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Obtain exons hg38 2 (Cluster data)\n",
    "\n",
    "exons2 = pd.read_csv(\"/workspace/projects/genomic_regions/hg38/release_2/cds_with_stop_codon/hg38_cds_with_stop_codon.canonical.overlap.gz\", sep=\"\\t\")\n",
    "exons2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "potential-voice",
   "metadata": {},
   "source": [
    "### OBTAIN EXONS FROM CH GENES\n",
    "\n",
    "**1. Prepare data**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "numerous-basement",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     DNMT3A\n",
       "1       TET2\n",
       "2      PPM1D\n",
       "3      ASXL1\n",
       "4        ATM\n",
       "       ...  \n",
       "60     KDM5C\n",
       "61     MYO5A\n",
       "62       ERF\n",
       "63      CUX1\n",
       "64    CDKN1B\n",
       "Name: Symbol, Length: 65, dtype: object"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Obtain list of CH genes\n",
    "\n",
    "ch_genes = ch_genes_df[\"Symbol\"]\n",
    "ch_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "unique-special",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Exon region start (bp)</th>\n",
       "      <th>Exon region end (bp)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>3307</td>\n",
       "      <td>4262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>4470</td>\n",
       "      <td>5511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>5904</td>\n",
       "      <td>7445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>7586</td>\n",
       "      <td>8269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>ENST00000361851</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>8366</td>\n",
       "      <td>8572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220303</th>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>ENST00000451251</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>248858131</td>\n",
       "      <td>248858321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220304</th>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>ENST00000451251</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>248850066</td>\n",
       "      <td>248850516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220305</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>ENST00000329291</td>\n",
       "      <td>1</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>248906235</td>\n",
       "      <td>248906342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220306</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>ENST00000329291</td>\n",
       "      <td>1</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>248913816</td>\n",
       "      <td>248913879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220307</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>ENST00000329291</td>\n",
       "      <td>1</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>248916602</td>\n",
       "      <td>248919146</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>220308 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene stable ID Transcript stable ID Chromosome/scaffold name  \\\n",
       "0       ENSG00000198888      ENST00000361390                       MT   \n",
       "1       ENSG00000198763      ENST00000361453                       MT   \n",
       "2       ENSG00000198804      ENST00000361624                       MT   \n",
       "3       ENSG00000198712      ENST00000361739                       MT   \n",
       "4       ENSG00000228253      ENST00000361851                       MT   \n",
       "...                 ...                  ...                      ...   \n",
       "220303  ENSG00000171163      ENST00000451251                        1   \n",
       "220304  ENSG00000171163      ENST00000451251                        1   \n",
       "220305  ENSG00000185220      ENST00000329291                        1   \n",
       "220306  ENSG00000185220      ENST00000329291                        1   \n",
       "220307  ENSG00000185220      ENST00000329291                        1   \n",
       "\n",
       "       Gene name  Exon region start (bp)  Exon region end (bp)  \n",
       "0         MT-ND1                    3307                  4262  \n",
       "1         MT-ND2                    4470                  5511  \n",
       "2         MT-CO1                    5904                  7445  \n",
       "3         MT-CO2                    7586                  8269  \n",
       "4        MT-ATP8                    8366                  8572  \n",
       "...          ...                     ...                   ...  \n",
       "220303    ZNF692               248858131             248858321  \n",
       "220304    ZNF692               248850066             248850516  \n",
       "220305     PGBD2               248906235             248906342  \n",
       "220306     PGBD2               248913816             248913879  \n",
       "220307     PGBD2               248916602             248919146  \n",
       "\n",
       "[220308 rows x 6 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Obtain reduced exons 1 data frame\n",
    "exons1_s = exons1.iloc[:, [0,2,4,7,8,9]]\n",
    "exons1_s"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "outside-publication",
   "metadata": {},
   "source": [
    "**2. Get exons CH genes Biomart**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "administrative-beauty",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Exon region start (bp)</th>\n",
       "      <th>Exon region end (bp)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15155</th>\n",
       "      <td>ENSG00000282932</td>\n",
       "      <td>ENST00000634556</td>\n",
       "      <td>CHR_HSCHR9_1_CTG6</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>10033718</td>\n",
       "      <td>10033790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15156</th>\n",
       "      <td>ENSG00000282932</td>\n",
       "      <td>ENST00000634556</td>\n",
       "      <td>CHR_HSCHR9_1_CTG6</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>9938507</td>\n",
       "      <td>9938610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15157</th>\n",
       "      <td>ENSG00000282932</td>\n",
       "      <td>ENST00000634556</td>\n",
       "      <td>CHR_HSCHR9_1_CTG6</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>9766810</td>\n",
       "      <td>9766851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15158</th>\n",
       "      <td>ENSG00000282932</td>\n",
       "      <td>ENST00000634556</td>\n",
       "      <td>CHR_HSCHR9_1_CTG6</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>9734533</td>\n",
       "      <td>9734571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15159</th>\n",
       "      <td>ENSG00000282932</td>\n",
       "      <td>ENST00000634556</td>\n",
       "      <td>CHR_HSCHR9_1_CTG6</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>9574732</td>\n",
       "      <td>9574781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217167</th>\n",
       "      <td>ENSG00000198625</td>\n",
       "      <td>ENST00000367182</td>\n",
       "      <td>1</td>\n",
       "      <td>MDM4</td>\n",
       "      <td>204538209</td>\n",
       "      <td>204538308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217168</th>\n",
       "      <td>ENSG00000198625</td>\n",
       "      <td>ENST00000367182</td>\n",
       "      <td>1</td>\n",
       "      <td>MDM4</td>\n",
       "      <td>204542784</td>\n",
       "      <td>204542944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217169</th>\n",
       "      <td>ENSG00000198625</td>\n",
       "      <td>ENST00000367182</td>\n",
       "      <td>1</td>\n",
       "      <td>MDM4</td>\n",
       "      <td>204544535</td>\n",
       "      <td>204544684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217170</th>\n",
       "      <td>ENSG00000198625</td>\n",
       "      <td>ENST00000367182</td>\n",
       "      <td>1</td>\n",
       "      <td>MDM4</td>\n",
       "      <td>204546797</td>\n",
       "      <td>204546877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217171</th>\n",
       "      <td>ENSG00000198625</td>\n",
       "      <td>ENST00000367182</td>\n",
       "      <td>1</td>\n",
       "      <td>MDM4</td>\n",
       "      <td>204549113</td>\n",
       "      <td>204558120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1171 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene stable ID Transcript stable ID Chromosome/scaffold name  \\\n",
       "15155   ENSG00000282932      ENST00000634556        CHR_HSCHR9_1_CTG6   \n",
       "15156   ENSG00000282932      ENST00000634556        CHR_HSCHR9_1_CTG6   \n",
       "15157   ENSG00000282932      ENST00000634556        CHR_HSCHR9_1_CTG6   \n",
       "15158   ENSG00000282932      ENST00000634556        CHR_HSCHR9_1_CTG6   \n",
       "15159   ENSG00000282932      ENST00000634556        CHR_HSCHR9_1_CTG6   \n",
       "...                 ...                  ...                      ...   \n",
       "217167  ENSG00000198625      ENST00000367182                        1   \n",
       "217168  ENSG00000198625      ENST00000367182                        1   \n",
       "217169  ENSG00000198625      ENST00000367182                        1   \n",
       "217170  ENSG00000198625      ENST00000367182                        1   \n",
       "217171  ENSG00000198625      ENST00000367182                        1   \n",
       "\n",
       "       Gene name  Exon region start (bp)  Exon region end (bp)  \n",
       "15155      PTPRD                10033718              10033790  \n",
       "15156      PTPRD                 9938507               9938610  \n",
       "15157      PTPRD                 9766810               9766851  \n",
       "15158      PTPRD                 9734533               9734571  \n",
       "15159      PTPRD                 9574732               9574781  \n",
       "...          ...                     ...                   ...  \n",
       "217167      MDM4               204538209             204538308  \n",
       "217168      MDM4               204542784             204542944  \n",
       "217169      MDM4               204544535             204544684  \n",
       "217170      MDM4               204546797             204546877  \n",
       "217171      MDM4               204549113             204558120  \n",
       "\n",
       "[1171 rows x 6 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Obtain exons from 65 genes\n",
    "\n",
    "ch_genes_exons1 = exons1_s[exons1_s[\"Gene name\"].isin(ch_genes)]\n",
    "\n",
    "ch_genes_exons1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "alpha-cancer",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "62"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check number of genes obtained\n",
    "len(ch_genes_exons1[\"Gene name\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "thorough-glossary",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15     LZTR1\n",
       "43    NOTCH1\n",
       "48      MKL1\n",
       "Name: Symbol, dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check missing genes\n",
    "ch_genes[-ch_genes.isin(ch_genes_exons1[\"Gene name\"])]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "defined-delivery",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save\n",
    "\n",
    "ch_genes_exons1.to_csv(\"/home/sdemajo/CH_gnomad_2021_02/results/CH_genes_exons_BMart.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "hollow-watson",
   "metadata": {},
   "source": [
    "**3. Get exons CH genes Cluster**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "literary-donor",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>GENE_ID</th>\n",
       "      <th>TRANSCRIPT_ID</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>1</td>\n",
       "      <td>1787331</td>\n",
       "      <td>1787437</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>1</td>\n",
       "      <td>1789053</td>\n",
       "      <td>1789269</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>1</td>\n",
       "      <td>1790395</td>\n",
       "      <td>1790596</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>1</td>\n",
       "      <td>1793245</td>\n",
       "      <td>1793311</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>1</td>\n",
       "      <td>1804419</td>\n",
       "      <td>1804581</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191001</th>\n",
       "      <td>X</td>\n",
       "      <td>124090575</td>\n",
       "      <td>124090764</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191002</th>\n",
       "      <td>X</td>\n",
       "      <td>124090854</td>\n",
       "      <td>124090964</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191003</th>\n",
       "      <td>X</td>\n",
       "      <td>124094018</td>\n",
       "      <td>124094144</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191004</th>\n",
       "      <td>X</td>\n",
       "      <td>124095372</td>\n",
       "      <td>124095449</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191005</th>\n",
       "      <td>X</td>\n",
       "      <td>124100574</td>\n",
       "      <td>124100597</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1133 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROMOSOME      START        END STRAND          GENE_ID  \\\n",
       "435             1    1787331    1787437      -  ENSG00000078369   \n",
       "436             1    1789053    1789269      -  ENSG00000078369   \n",
       "437             1    1790395    1790596      -  ENSG00000078369   \n",
       "438             1    1793245    1793311      -  ENSG00000078369   \n",
       "439             1    1804419    1804581      -  ENSG00000078369   \n",
       "...           ...        ...        ...    ...              ...   \n",
       "191001          X  124090575  124090764      +  ENSG00000101972   \n",
       "191002          X  124090854  124090964      +  ENSG00000101972   \n",
       "191003          X  124094018  124094144      +  ENSG00000101972   \n",
       "191004          X  124095372  124095449      +  ENSG00000101972   \n",
       "191005          X  124100574  124100597      +  ENSG00000101972   \n",
       "\n",
       "          TRANSCRIPT_ID SYMBOL  \n",
       "435     ENST00000378609   GNB1  \n",
       "436     ENST00000378609   GNB1  \n",
       "437     ENST00000378609   GNB1  \n",
       "438     ENST00000378609   GNB1  \n",
       "439     ENST00000378609   GNB1  \n",
       "...                 ...    ...  \n",
       "191001  ENST00000371145  STAG2  \n",
       "191002  ENST00000371145  STAG2  \n",
       "191003  ENST00000371145  STAG2  \n",
       "191004  ENST00000371145  STAG2  \n",
       "191005  ENST00000371145  STAG2  \n",
       "\n",
       "[1133 rows x 7 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Obtain exons from 65 genes\n",
    "\n",
    "ch_genes_exons2 = exons2[exons2[\"SYMBOL\"].isin(ch_genes)]\n",
    "\n",
    "ch_genes_exons2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "honey-brunei",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check number of genes obtained\n",
    "len(ch_genes_exons2[\"SYMBOL\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "minus-endorsement",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "48    MKL1\n",
       "Name: Symbol, dtype: object"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check missing genes\n",
    "ch_genes[-ch_genes.isin(ch_genes_exons2[\"SYMBOL\"])]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "dutch-printing",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>GENE_ID</th>\n",
       "      <th>TRANSCRIPT_ID</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>1</td>\n",
       "      <td>1787331</td>\n",
       "      <td>1787437</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>1</td>\n",
       "      <td>1789053</td>\n",
       "      <td>1789269</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>1</td>\n",
       "      <td>1790395</td>\n",
       "      <td>1790596</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>1</td>\n",
       "      <td>1793245</td>\n",
       "      <td>1793311</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>1</td>\n",
       "      <td>1804419</td>\n",
       "      <td>1804581</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191001</th>\n",
       "      <td>X</td>\n",
       "      <td>124090575</td>\n",
       "      <td>124090764</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191002</th>\n",
       "      <td>X</td>\n",
       "      <td>124090854</td>\n",
       "      <td>124090964</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191003</th>\n",
       "      <td>X</td>\n",
       "      <td>124094018</td>\n",
       "      <td>124094144</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191004</th>\n",
       "      <td>X</td>\n",
       "      <td>124095372</td>\n",
       "      <td>124095449</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191005</th>\n",
       "      <td>X</td>\n",
       "      <td>124100574</td>\n",
       "      <td>124100597</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1146 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROMOSOME      START        END STRAND          GENE_ID  \\\n",
       "435             1    1787331    1787437      -  ENSG00000078369   \n",
       "436             1    1789053    1789269      -  ENSG00000078369   \n",
       "437             1    1790395    1790596      -  ENSG00000078369   \n",
       "438             1    1793245    1793311      -  ENSG00000078369   \n",
       "439             1    1804419    1804581      -  ENSG00000078369   \n",
       "...           ...        ...        ...    ...              ...   \n",
       "191001          X  124090575  124090764      +  ENSG00000101972   \n",
       "191002          X  124090854  124090964      +  ENSG00000101972   \n",
       "191003          X  124094018  124094144      +  ENSG00000101972   \n",
       "191004          X  124095372  124095449      +  ENSG00000101972   \n",
       "191005          X  124100574  124100597      +  ENSG00000101972   \n",
       "\n",
       "          TRANSCRIPT_ID SYMBOL  \n",
       "435     ENST00000378609   GNB1  \n",
       "436     ENST00000378609   GNB1  \n",
       "437     ENST00000378609   GNB1  \n",
       "438     ENST00000378609   GNB1  \n",
       "439     ENST00000378609   GNB1  \n",
       "...                 ...    ...  \n",
       "191001  ENST00000371145  STAG2  \n",
       "191002  ENST00000371145  STAG2  \n",
       "191003  ENST00000371145  STAG2  \n",
       "191004  ENST00000371145  STAG2  \n",
       "191005  ENST00000371145  STAG2  \n",
       "\n",
       "[1146 rows x 7 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Obtain genes again with MKL1 (=MRTFA)\n",
    "ch_genes = ch_genes.append(pd.Series(\"MRTFA\"))\n",
    "ch_genes_exons2 = exons2[exons2[\"SYMBOL\"].isin(ch_genes)]\n",
    "\n",
    "ch_genes_exons2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "brave-allen",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "65"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check number of genes obtained\n",
    "len(ch_genes_exons2[\"SYMBOL\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "apparent-breakdown",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save\n",
    "\n",
    "ch_genes_exons2.to_csv(\"/home/sdemajo/CH_gnomad_2021_02/results/CH_genes_exons_Clust.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fresh-africa",
   "metadata": {},
   "source": [
    "### Compare results from the two sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "metallic-accessory",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMart</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PTPRD</th>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATM</th>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2C</th>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NF1</th>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2D</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMEM127</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDHAF2</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDKN1B</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYCN</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRSF2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         BioMart\n",
       "PTPRD         86\n",
       "ATM           63\n",
       "KMT2C         59\n",
       "NF1           58\n",
       "KMT2D         55\n",
       "...          ...\n",
       "TMEM127        4\n",
       "SDHAF2         4\n",
       "CDKN1B         3\n",
       "MYCN           3\n",
       "SRSF2          2\n",
       "\n",
       "[62 rows x 1 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Check exons for each gene\n",
    "\n",
    "# BioMart data\n",
    "#exonresults1 = ch_genes_exons1.groupby(\"Gene name\").count().iloc[:,1]\n",
    "exonresults1 = ch_genes_exons1['Gene name'].value_counts().to_frame()\n",
    "exonresults1.columns = [\"BioMart\"] # Change column name\n",
    "exonresults1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "mexican-bookmark",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ATM</th>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2C</th>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NF1</th>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2D</th>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYO5A</th>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NRAS</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMEM127</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRSF2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDKN1B</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYCN</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Cluster\n",
       "ATM           62\n",
       "KMT2C         59\n",
       "NF1           58\n",
       "KMT2D         54\n",
       "MYO5A         41\n",
       "...          ...\n",
       "NRAS           4\n",
       "TMEM127        3\n",
       "SRSF2          2\n",
       "CDKN1B         2\n",
       "MYCN           2\n",
       "\n",
       "[65 rows x 1 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Cluster data\n",
    "\n",
    "#exonresults2 = ch_genes_exons2.groupby(\"SYMBOL\").count().iloc[:,0]\n",
    "\n",
    "exonresults2 = ch_genes_exons2['SYMBOL'].value_counts().to_frame()\n",
    "exonresults2.columns = [\"Cluster\"] # Change column name\n",
    "exonresults2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "qualified-basis",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cluster</th>\n",
       "      <th>BioMart</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ATM</th>\n",
       "      <td>62</td>\n",
       "      <td>63.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2C</th>\n",
       "      <td>59</td>\n",
       "      <td>59.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NF1</th>\n",
       "      <td>58</td>\n",
       "      <td>58.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2D</th>\n",
       "      <td>54</td>\n",
       "      <td>55.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYO5A</th>\n",
       "      <td>41</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTPRD</th>\n",
       "      <td>35</td>\n",
       "      <td>86.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH1</th>\n",
       "      <td>34</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAG2</th>\n",
       "      <td>33</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDM6A</th>\n",
       "      <td>29</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDM5C</th>\n",
       "      <td>26</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SF3B1</th>\n",
       "      <td>25</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUX1</th>\n",
       "      <td>24</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JAK2</th>\n",
       "      <td>23</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAT3</th>\n",
       "      <td>23</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGA</th>\n",
       "      <td>23</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFF3</th>\n",
       "      <td>23</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERCC2</th>\n",
       "      <td>23</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNMT3B</th>\n",
       "      <td>22</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNMT3A</th>\n",
       "      <td>22</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARID2</th>\n",
       "      <td>21</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LZTR1</th>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNM2</th>\n",
       "      <td>21</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RET</th>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EZH2</th>\n",
       "      <td>19</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAT5B</th>\n",
       "      <td>18</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EPHA3</th>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CBL</th>\n",
       "      <td>16</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FOXP1</th>\n",
       "      <td>16</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SUZ12</th>\n",
       "      <td>16</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC</th>\n",
       "      <td>15</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHEK2</th>\n",
       "      <td>15</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTPN11</th>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP63</th>\n",
       "      <td>14</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PABPC1</th>\n",
       "      <td>14</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAD21</th>\n",
       "      <td>13</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GNAS</th>\n",
       "      <td>13</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASXL1</th>\n",
       "      <td>13</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRTFA</th>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MPL</th>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATE1</th>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABL2</th>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZRSR2</th>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDH2</th>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTCF</th>\n",
       "      <td>10</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>10</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MDM4</th>\n",
       "      <td>10</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GNB1</th>\n",
       "      <td>9</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CALR</th>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TET2</th>\n",
       "      <td>9</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RUNX1</th>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>U2AF1</th>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDH1</th>\n",
       "      <td>8</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AR</th>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPARG</th>\n",
       "      <td>7</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SH2B3</th>\n",
       "      <td>7</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPM1D</th>\n",
       "      <td>6</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYD88</th>\n",
       "      <td>5</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDHAF2</th>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERF</th>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KRAS</th>\n",
       "      <td>4</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NRAS</th>\n",
       "      <td>4</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMEM127</th>\n",
       "      <td>3</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRSF2</th>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDKN1B</th>\n",
       "      <td>2</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYCN</th>\n",
       "      <td>2</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Cluster  BioMart\n",
       "ATM           62     63.0\n",
       "KMT2C         59     59.0\n",
       "NF1           58     58.0\n",
       "KMT2D         54     55.0\n",
       "MYO5A         41     41.0\n",
       "PTPRD         35     86.0\n",
       "NOTCH1        34      NaN\n",
       "STAG2         33     35.0\n",
       "KDM6A         29     29.0\n",
       "KDM5C         26     26.0\n",
       "SF3B1         25     25.0\n",
       "CUX1          24     24.0\n",
       "JAK2          23     25.0\n",
       "STAT3         23     24.0\n",
       "MGA           23     24.0\n",
       "AFF3          23     25.0\n",
       "ERCC2         23     23.0\n",
       "DNMT3B        22     23.0\n",
       "DNMT3A        22     23.0\n",
       "ARID2         21     21.0\n",
       "LZTR1         21      NaN\n",
       "DNM2          21     21.0\n",
       "RET           20     20.0\n",
       "EZH2          19     20.0\n",
       "STAT5B        18     19.0\n",
       "EPHA3         17     17.0\n",
       "CBL           16     16.0\n",
       "FOXP1         16     21.0\n",
       "SUZ12         16     16.0\n",
       "APC           15     16.0\n",
       "CHEK2         15     16.0\n",
       "PTPN11        15     15.0\n",
       "TP63          14     14.0\n",
       "PABPC1        14     15.0\n",
       "RAD21         13     14.0\n",
       "GNAS          13     13.0\n",
       "ASXL1         13     13.0\n",
       "MRTFA         13      NaN\n",
       "MPL           12     12.0\n",
       "ATE1          12     12.0\n",
       "ABL2          12     12.0\n",
       "ZRSR2         11     11.0\n",
       "IDH2          11     11.0\n",
       "CTCF          10     13.0\n",
       "TP53          10     11.0\n",
       "MDM4          10     11.0\n",
       "GNB1           9     11.0\n",
       "CALR           9      9.0\n",
       "TET2           9     11.0\n",
       "RUNX1          8      8.0\n",
       "U2AF1          8      8.0\n",
       "IDH1           8     10.0\n",
       "AR             8      8.0\n",
       "PPARG          7      7.0\n",
       "SH2B3          7      8.0\n",
       "PPM1D          6      6.0\n",
       "MYD88          5      4.0\n",
       "SDHAF2         4      4.0\n",
       "ERF            4      4.0\n",
       "KRAS           4      6.0\n",
       "NRAS           4      7.0\n",
       "TMEM127        3      4.0\n",
       "SRSF2          2      2.0\n",
       "CDKN1B         2      3.0\n",
       "MYCN           2      3.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Compare both results\n",
    "\n",
    "# Join data\n",
    "exonresults12 = exonresults2.join(exonresults1)\n",
    "\n",
    "# show all results\n",
    "with pd.option_context(\"display.max_rows\", 100):\n",
    "    display(exonresults12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "proud-design",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cluster    1146.0\n",
       "BioMart    1171.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get total exon number\n",
    "\n",
    "exonresults12.sum()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
