{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "republican-station",
   "metadata": {},
   "source": [
    "## Obtain gnomAD variants corresponding to the 65 CH genes\n",
    "\n",
    "### CH gene exons coordenates\n",
    "\n",
    "Corresponding to: \"CH_genes_exons_Clust.tsv\"\n",
    "\n",
    "Obtained by notebook: **\"CH_genes_exons\"**\n",
    "\n",
    "\n",
    "### Variant data\n",
    "\n",
    "Obtained from: \"/workspace/datasets/gnomad/data/v2.1.1/exomes/hg38/gnomad.exomes.r2.1.1.sites.liftover_grch38.vcf.bgz\"\n",
    "\n",
    "gnomAD v2.1.1, taking only exons:\n",
    "\n",
    "- total: 125,748 samples\n",
    "\n",
    "- non-cancer: 118,479 samples\n",
    "\n",
    "- controls/biobank: 54,704 samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "broadband-cabin",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "confidential-testing",
   "metadata": {},
   "outputs": [],
   "source": [
    "### GET ALL VARIANTS FROM ALL CH GENES EXONS\n",
    "\n",
    "### Open CH genes exon coordenates\n",
    "# Open file\n",
    "f = open(\"/home/sdemajo/CH_gnomad_2021_02/results/CH_genes_exons_Clust.tsv\")\n",
    "# Read lines skipping the first\n",
    "ch_exons = f.readlines()[1:]\n",
    "# Close file\n",
    "f.close()\n",
    "\n",
    "### Creat tabix object with TabixFile\n",
    "# (allows fast random access to compressed and tabix indexed tab-separated file formats with genomic data)\n",
    "file = \"/workspace/datasets/gnomad/data/v2.1.1/exomes/hg38/gnomad.exomes.r2.1.1.sites.liftover_grch38.vcf.bgz\"\n",
    "tbx = pysam.TabixFile(file)\n",
    "\n",
    "### Obtain variant data\n",
    "\n",
    "# Create an empty list for all variants\n",
    "variant_all = []\n",
    "\n",
    "# Iterate every exon position\n",
    "for n, line in enumerate(ch_exons):\n",
    "    # Split data tab\n",
    "    exon_data = line.split(\"\\t\")\n",
    "    # Obtain localization and gene info for every exon\n",
    "    chrom = \"chr\" + exon_data[1]\n",
    "    start = int(exon_data[2])\n",
    "    end = int(exon_data[3])\n",
    "    gene = exon_data[7].replace('\\n','')\n",
    "    transcript = exon_data[6].replace('\\n','')\n",
    "\n",
    "    # Create an empty list for variants from this exon\n",
    "    variant_list = []\n",
    "\n",
    "    # Fetch to iterate over all of the read mapping to a specified region\n",
    "    for row in tbx.fetch(chrom, start, end):\n",
    "\n",
    "        # get variant data (except last column)\n",
    "        var_info = row.split(\"\\t\")[0:-1]\n",
    "\n",
    "        # add gene name and transcript code\n",
    "        var_info.append(gene)\n",
    "        var_info.append(transcript)\n",
    "\n",
    "        # get AF: allel frequency\n",
    "        var_info2 = row.split(\"\\t\")[-1] # get last column (INFO)\n",
    "        var_info2 = var_info2.split(\";\") # split by ;\n",
    "        # iterate through INFO data and extract AF\n",
    "        for i in var_info2:\n",
    "            if i[0:3] == \"AF=\":\n",
    "                var_info.append(i)\n",
    "            elif i[0:12] == \"controls_AF=\":\n",
    "                var_info.append(i)\n",
    "            elif i[0:14] == \"non_cancer_AF=\":\n",
    "                var_info.append(i)\n",
    "\n",
    "        # Add variant to the exon list\n",
    "        variant_list.append(var_info)\n",
    "\n",
    "    # Add \n",
    "    variant_all.append(variant_list)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "egyptian-cooling",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'list'>\n",
      "1146\n"
     ]
    }
   ],
   "source": [
    "### Check results\n",
    "print(type(variant_all))\n",
    "print(len(variant_all))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "collect-gateway",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>AF</th>\n",
       "      <th>controls_AF</th>\n",
       "      <th>non_cancer_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787332</td>\n",
       "      <td>rs747120803</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4570.98</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>AF=1.59937e-05</td>\n",
       "      <td>controls_AF=9.18560e-06</td>\n",
       "      <td>non_cancer_AF=1.69804e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787334</td>\n",
       "      <td>rs1362537090</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>314.39</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>AF=3.99505e-06</td>\n",
       "      <td>controls_AF=9.17852e-06</td>\n",
       "      <td>non_cancer_AF=4.24128e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787357</td>\n",
       "      <td>rs1386056041</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>372.36</td>\n",
       "      <td>AC0;RF</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>AF=0.00000e+00</td>\n",
       "      <td>controls_AF=0.00000e+00</td>\n",
       "      <td>non_cancer_AF=0.00000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787367</td>\n",
       "      <td>rs750998849</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1667.37</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>AF=3.98238e-06</td>\n",
       "      <td>controls_AF=0.00000e+00</td>\n",
       "      <td>non_cancer_AF=4.22697e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787370</td>\n",
       "      <td>rs139152573</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>7962.49</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>AF=3.58403e-05</td>\n",
       "      <td>controls_AF=1.83046e-05</td>\n",
       "      <td>non_cancer_AF=2.95878e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54150</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095410</td>\n",
       "      <td>rs750598294</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1182.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>AF=5.45461e-06</td>\n",
       "      <td>controls_AF=1.24504e-05</td>\n",
       "      <td>non_cancer_AF=5.80966e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54151</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095442</td>\n",
       "      <td>rs1230398678</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>3145.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>AF=5.46033e-06</td>\n",
       "      <td>controls_AF=0.00000e+00</td>\n",
       "      <td>non_cancer_AF=5.81564e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54152</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095446</td>\n",
       "      <td>rs201262699</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>53165.35</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>AF=1.80286e-04</td>\n",
       "      <td>controls_AF=1.99414e-04</td>\n",
       "      <td>non_cancer_AF=1.57104e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54153</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100577</td>\n",
       "      <td>rs1304348952</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1266.82</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>AF=5.53710e-06</td>\n",
       "      <td>controls_AF=1.26076e-05</td>\n",
       "      <td>non_cancer_AF=5.90099e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54154</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100582</td>\n",
       "      <td>rs1371148856</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>142.99</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>AF=5.53986e-06</td>\n",
       "      <td>controls_AF=1.26178e-05</td>\n",
       "      <td>non_cancer_AF=5.90409e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54155 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chromosome   Position          rsID REF ALT      QUAL  FILTER   Gene  \\\n",
       "0           chr1    1787332   rs747120803   T   C   4570.98    PASS   GNB1   \n",
       "1           chr1    1787334  rs1362537090   G   A    314.39    PASS   GNB1   \n",
       "2           chr1    1787357  rs1386056041   C   T    372.36  AC0;RF   GNB1   \n",
       "3           chr1    1787367   rs750998849   T   C   1667.37    PASS   GNB1   \n",
       "4           chr1    1787370   rs139152573   C   T   7962.49    PASS   GNB1   \n",
       "...          ...        ...           ...  ..  ..       ...     ...    ...   \n",
       "54150       chrX  124095410   rs750598294   G   A   1182.36    PASS  STAG2   \n",
       "54151       chrX  124095442  rs1230398678   A   T   3145.36    PASS  STAG2   \n",
       "54152       chrX  124095446   rs201262699   A   T  53165.35    PASS  STAG2   \n",
       "54153       chrX  124100577  rs1304348952   C   A   1266.82    PASS  STAG2   \n",
       "54154       chrX  124100582  rs1371148856   A   T    142.99    PASS  STAG2   \n",
       "\n",
       "            Transcript              AF              controls_AF  \\\n",
       "0      ENST00000378609  AF=1.59937e-05  controls_AF=9.18560e-06   \n",
       "1      ENST00000378609  AF=3.99505e-06  controls_AF=9.17852e-06   \n",
       "2      ENST00000378609  AF=0.00000e+00  controls_AF=0.00000e+00   \n",
       "3      ENST00000378609  AF=3.98238e-06  controls_AF=0.00000e+00   \n",
       "4      ENST00000378609  AF=3.58403e-05  controls_AF=1.83046e-05   \n",
       "...                ...             ...                      ...   \n",
       "54150  ENST00000371145  AF=5.45461e-06  controls_AF=1.24504e-05   \n",
       "54151  ENST00000371145  AF=5.46033e-06  controls_AF=0.00000e+00   \n",
       "54152  ENST00000371145  AF=1.80286e-04  controls_AF=1.99414e-04   \n",
       "54153  ENST00000371145  AF=5.53710e-06  controls_AF=1.26076e-05   \n",
       "54154  ENST00000371145  AF=5.53986e-06  controls_AF=1.26178e-05   \n",
       "\n",
       "                   non_cancer_AF  \n",
       "0      non_cancer_AF=1.69804e-05  \n",
       "1      non_cancer_AF=4.24128e-06  \n",
       "2      non_cancer_AF=0.00000e+00  \n",
       "3      non_cancer_AF=4.22697e-06  \n",
       "4      non_cancer_AF=2.95878e-05  \n",
       "...                          ...  \n",
       "54150  non_cancer_AF=5.80966e-06  \n",
       "54151  non_cancer_AF=5.81564e-06  \n",
       "54152  non_cancer_AF=1.57104e-04  \n",
       "54153  non_cancer_AF=5.90099e-06  \n",
       "54154  non_cancer_AF=5.90409e-06  \n",
       "\n",
       "[54155 rows x 12 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### TRANSFORM RESULTS TO DATA FRAME\n",
    "\n",
    "## 1. Transform list of lists to one list\n",
    "\n",
    "ch_variants = [item for sublist in variant_all for item in sublist]\n",
    "\n",
    "\n",
    "## 2. Transform list (of lists) to data frame\n",
    "ch_variants = pd.DataFrame(ch_variants, columns = [\"Chromosome\", \"Position\", \"rsID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\",\n",
    "                                         \"Gene\", \"Transcript\", \"AF\", \"controls_AF\", \"non_cancer_AF\"])\n",
    "ch_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "demographic-activity",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chromosome        object\n",
      "Position          object\n",
      "rsID              object\n",
      "REF               object\n",
      "ALT               object\n",
      "QUAL              object\n",
      "FILTER            object\n",
      "Gene              object\n",
      "Transcript        object\n",
      "AF               float64\n",
      "controls_AF      float64\n",
      "non_cancer_AF    float64\n",
      "dtype: object\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>AF</th>\n",
       "      <th>controls_AF</th>\n",
       "      <th>non_cancer_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787332</td>\n",
       "      <td>rs747120803</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4570.98</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787334</td>\n",
       "      <td>rs1362537090</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>314.39</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787357</td>\n",
       "      <td>rs1386056041</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>372.36</td>\n",
       "      <td>AC0;RF</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787367</td>\n",
       "      <td>rs750998849</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1667.37</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787370</td>\n",
       "      <td>rs139152573</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>7962.49</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.000030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54150</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095410</td>\n",
       "      <td>rs750598294</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1182.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54151</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095442</td>\n",
       "      <td>rs1230398678</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>3145.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54152</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095446</td>\n",
       "      <td>rs201262699</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>53165.35</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000180</td>\n",
       "      <td>0.000199</td>\n",
       "      <td>0.000157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54153</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100577</td>\n",
       "      <td>rs1304348952</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1266.82</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54154</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100582</td>\n",
       "      <td>rs1371148856</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>142.99</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54155 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chromosome   Position          rsID REF ALT      QUAL  FILTER   Gene  \\\n",
       "0           chr1    1787332   rs747120803   T   C   4570.98    PASS   GNB1   \n",
       "1           chr1    1787334  rs1362537090   G   A    314.39    PASS   GNB1   \n",
       "2           chr1    1787357  rs1386056041   C   T    372.36  AC0;RF   GNB1   \n",
       "3           chr1    1787367   rs750998849   T   C   1667.37    PASS   GNB1   \n",
       "4           chr1    1787370   rs139152573   C   T   7962.49    PASS   GNB1   \n",
       "...          ...        ...           ...  ..  ..       ...     ...    ...   \n",
       "54150       chrX  124095410   rs750598294   G   A   1182.36    PASS  STAG2   \n",
       "54151       chrX  124095442  rs1230398678   A   T   3145.36    PASS  STAG2   \n",
       "54152       chrX  124095446   rs201262699   A   T  53165.35    PASS  STAG2   \n",
       "54153       chrX  124100577  rs1304348952   C   A   1266.82    PASS  STAG2   \n",
       "54154       chrX  124100582  rs1371148856   A   T    142.99    PASS  STAG2   \n",
       "\n",
       "            Transcript        AF  controls_AF  non_cancer_AF  \n",
       "0      ENST00000378609  0.000016     0.000009       0.000017  \n",
       "1      ENST00000378609  0.000004     0.000009       0.000004  \n",
       "2      ENST00000378609  0.000000     0.000000       0.000000  \n",
       "3      ENST00000378609  0.000004     0.000000       0.000004  \n",
       "4      ENST00000378609  0.000036     0.000018       0.000030  \n",
       "...                ...       ...          ...            ...  \n",
       "54150  ENST00000371145  0.000005     0.000012       0.000006  \n",
       "54151  ENST00000371145  0.000005     0.000000       0.000006  \n",
       "54152  ENST00000371145  0.000180     0.000199       0.000157  \n",
       "54153  ENST00000371145  0.000006     0.000013       0.000006  \n",
       "54154  ENST00000371145  0.000006     0.000013       0.000006  \n",
       "\n",
       "[54155 rows x 12 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### MODIFY DATA FRAME\n",
    "\n",
    "# Eliminate \"AF=\" text\n",
    "ch_variants[\"AF\"] = ch_variants[\"AF\"].str.replace(\"AF=\", \"\")\n",
    "ch_variants[\"controls_AF\"] = ch_variants[\"controls_AF\"].str.replace(\"controls_AF=\", \"\")\n",
    "ch_variants[\"non_cancer_AF\"] = ch_variants[\"non_cancer_AF\"].str.replace(\"non_cancer_AF=\", \"\")\n",
    "\n",
    "# Transform to number\n",
    "ch_variants[\"AF\"] = pd.to_numeric(ch_variants[\"AF\"])\n",
    "ch_variants[\"controls_AF\"] = pd.to_numeric(ch_variants[\"controls_AF\"])\n",
    "ch_variants[\"non_cancer_AF\"] = pd.to_numeric(ch_variants[\"non_cancer_AF\"])\n",
    "\n",
    "print(ch_variants.dtypes)\n",
    "ch_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "current-burst",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PASS                      47956\n",
      "AC0;RF                     2587\n",
      "AC0                        1861\n",
      "RF                         1721\n",
      "InbreedingCoeff;RF           14\n",
      "InbreedingCoeff              13\n",
      "AC0;InbreedingCoeff           2\n",
      "AC0;InbreedingCoeff;RF        1\n",
      "Name: FILTER, dtype: int64\n",
      "\n",
      "PASS    47956\n",
      "Name: FILTER, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "### FILTER VARIANTS THAT DO NOT PASS QC\n",
    "\n",
    "# Check FILTER variable\n",
    "print(ch_variants[\"FILTER\"].value_counts())\n",
    "print()\n",
    "\n",
    "# Select only good QC\n",
    "ch_variants = ch_variants[ch_variants[\"FILTER\"] == \"PASS\"]\n",
    "print(ch_variants[\"FILTER\"].value_counts()) # Check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "adapted-yeast",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>AF</th>\n",
       "      <th>controls_AF</th>\n",
       "      <th>non_cancer_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787332</td>\n",
       "      <td>rs747120803</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4570.98</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787334</td>\n",
       "      <td>rs1362537090</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>314.39</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787367</td>\n",
       "      <td>rs750998849</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1667.37</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787370</td>\n",
       "      <td>rs139152573</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>7962.49</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.000030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1787388</td>\n",
       "      <td>rs766891832</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2299.37</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>ENST00000378609</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54150</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095410</td>\n",
       "      <td>rs750598294</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1182.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54151</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095442</td>\n",
       "      <td>rs1230398678</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>3145.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54152</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095446</td>\n",
       "      <td>rs201262699</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>53165.35</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000180</td>\n",
       "      <td>0.000199</td>\n",
       "      <td>0.000157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54153</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100577</td>\n",
       "      <td>rs1304348952</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1266.82</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54154</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100582</td>\n",
       "      <td>rs1371148856</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>142.99</td>\n",
       "      <td>PASS</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000006</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47956 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chromosome   Position          rsID REF ALT      QUAL FILTER   Gene  \\\n",
       "0           chr1    1787332   rs747120803   T   C   4570.98   PASS   GNB1   \n",
       "1           chr1    1787334  rs1362537090   G   A    314.39   PASS   GNB1   \n",
       "3           chr1    1787367   rs750998849   T   C   1667.37   PASS   GNB1   \n",
       "4           chr1    1787370   rs139152573   C   T   7962.49   PASS   GNB1   \n",
       "6           chr1    1787388   rs766891832   G   A   2299.37   PASS   GNB1   \n",
       "...          ...        ...           ...  ..  ..       ...    ...    ...   \n",
       "54150       chrX  124095410   rs750598294   G   A   1182.36   PASS  STAG2   \n",
       "54151       chrX  124095442  rs1230398678   A   T   3145.36   PASS  STAG2   \n",
       "54152       chrX  124095446   rs201262699   A   T  53165.35   PASS  STAG2   \n",
       "54153       chrX  124100577  rs1304348952   C   A   1266.82   PASS  STAG2   \n",
       "54154       chrX  124100582  rs1371148856   A   T    142.99   PASS  STAG2   \n",
       "\n",
       "            Transcript        AF  controls_AF  non_cancer_AF  \n",
       "0      ENST00000378609  0.000016     0.000009       0.000017  \n",
       "1      ENST00000378609  0.000004     0.000009       0.000004  \n",
       "3      ENST00000378609  0.000004     0.000000       0.000004  \n",
       "4      ENST00000378609  0.000036     0.000018       0.000030  \n",
       "6      ENST00000378609  0.000004     0.000000       0.000004  \n",
       "...                ...       ...          ...            ...  \n",
       "54150  ENST00000371145  0.000005     0.000012       0.000006  \n",
       "54151  ENST00000371145  0.000005     0.000000       0.000006  \n",
       "54152  ENST00000371145  0.000180     0.000199       0.000157  \n",
       "54153  ENST00000371145  0.000006     0.000013       0.000006  \n",
       "54154  ENST00000371145  0.000006     0.000013       0.000006  \n",
       "\n",
       "[47956 rows x 12 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### SAVE\n",
    "\n",
    "ch_variants.to_csv(\"/home/sdemajo/CH_gnomad_2021_02/results/CH_genes_variants.tsv\", sep=\"\\t\")\n",
    "\n",
    "ch_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "equal-teacher",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KMT2D     4141\n",
       "KMT2C     3272\n",
       "NOTCH1    2296\n",
       "ATM       2211\n",
       "APC       2090\n",
       "          ... \n",
       "GNB1       158\n",
       "SDHAF2     142\n",
       "U2AF1      106\n",
       "NRAS        99\n",
       "KRAS        72\n",
       "Name: Gene, Length: 65, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### STUDY RESULTS\n",
    "\n",
    "## 1. Number of variants for each gene\n",
    "# (Non-normalized by size)\n",
    "ch_variants[\"Gene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "experienced-punishment",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>AF</th>\n",
       "      <th>controls_AF</th>\n",
       "      <th>non_cancer_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21631</th>\n",
       "      <td>chr17</td>\n",
       "      <td>76736425</td>\n",
       "      <td>rs237058</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>446471437.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SRSF2</td>\n",
       "      <td>ENST00000392485</td>\n",
       "      <td>0.997716</td>\n",
       "      <td>0.997639</td>\n",
       "      <td>0.997741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17209</th>\n",
       "      <td>chr15</td>\n",
       "      <td>52397434</td>\n",
       "      <td>rs1724577</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>686821345.29</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MYO5A</td>\n",
       "      <td>ENST00000399231</td>\n",
       "      <td>0.968857</td>\n",
       "      <td>0.965944</td>\n",
       "      <td>0.968792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16892</th>\n",
       "      <td>chr15</td>\n",
       "      <td>52375355</td>\n",
       "      <td>rs2414145</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>660859405.83</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MYO5A</td>\n",
       "      <td>ENST00000399231</td>\n",
       "      <td>0.959533</td>\n",
       "      <td>0.955890</td>\n",
       "      <td>0.959584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3400</th>\n",
       "      <td>chr10</td>\n",
       "      <td>121913824</td>\n",
       "      <td>rs10749435</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>300183430.26</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ATE1</td>\n",
       "      <td>ENST00000224652</td>\n",
       "      <td>0.899615</td>\n",
       "      <td>0.894335</td>\n",
       "      <td>0.898484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3158</th>\n",
       "      <td>chr10</td>\n",
       "      <td>121836739</td>\n",
       "      <td>rs4237536</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>359903111.56</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ATE1</td>\n",
       "      <td>ENST00000224652</td>\n",
       "      <td>0.895802</td>\n",
       "      <td>0.891226</td>\n",
       "      <td>0.894579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21695</th>\n",
       "      <td>chr17</td>\n",
       "      <td>76737017</td>\n",
       "      <td>rs237057</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>374166392.60</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SRSF2</td>\n",
       "      <td>ENST00000392485</td>\n",
       "      <td>0.894510</td>\n",
       "      <td>0.890979</td>\n",
       "      <td>0.894797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14484</th>\n",
       "      <td>chr15</td>\n",
       "      <td>41699117</td>\n",
       "      <td>rs2178004</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>247403229.11</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MGA</td>\n",
       "      <td>ENST00000219905</td>\n",
       "      <td>0.822932</td>\n",
       "      <td>0.822809</td>\n",
       "      <td>0.824158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14881</th>\n",
       "      <td>chr15</td>\n",
       "      <td>41734566</td>\n",
       "      <td>rs1918314</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>174505923.81</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MGA</td>\n",
       "      <td>ENST00000219905</td>\n",
       "      <td>0.800069</td>\n",
       "      <td>0.798792</td>\n",
       "      <td>0.800607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39439</th>\n",
       "      <td>chr5</td>\n",
       "      <td>112841059</td>\n",
       "      <td>rs459552</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>493715924.54</td>\n",
       "      <td>PASS</td>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>0.794825</td>\n",
       "      <td>0.793104</td>\n",
       "      <td>0.794920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15384</th>\n",
       "      <td>chr15</td>\n",
       "      <td>41749514</td>\n",
       "      <td>rs2695163</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>357988193.93</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MGA</td>\n",
       "      <td>ENST00000219905</td>\n",
       "      <td>0.792394</td>\n",
       "      <td>0.790177</td>\n",
       "      <td>0.793146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2744</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43118395</td>\n",
       "      <td>rs1800861</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>477129802.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>RET</td>\n",
       "      <td>ENST00000355710</td>\n",
       "      <td>0.739213</td>\n",
       "      <td>0.736804</td>\n",
       "      <td>0.738419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2045</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43100520</td>\n",
       "      <td>rs1800858</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>419195220.91</td>\n",
       "      <td>PASS</td>\n",
       "      <td>RET</td>\n",
       "      <td>ENST00000355710</td>\n",
       "      <td>0.731156</td>\n",
       "      <td>0.726512</td>\n",
       "      <td>0.730512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2420</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43111239</td>\n",
       "      <td>rs1800860</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>609661534.27</td>\n",
       "      <td>PASS</td>\n",
       "      <td>RET</td>\n",
       "      <td>ENST00000355710</td>\n",
       "      <td>0.695901</td>\n",
       "      <td>0.694847</td>\n",
       "      <td>0.695013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18485</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676154</td>\n",
       "      <td>rs1042522</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>716976924.97</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>0.668165</td>\n",
       "      <td>0.661152</td>\n",
       "      <td>0.666799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13344</th>\n",
       "      <td>chr12</td>\n",
       "      <td>111446804</td>\n",
       "      <td>rs3184504</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>147656611.61</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SH2B3</td>\n",
       "      <td>ENST00000341259</td>\n",
       "      <td>0.667391</td>\n",
       "      <td>0.699269</td>\n",
       "      <td>0.673459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27363</th>\n",
       "      <td>chr2</td>\n",
       "      <td>197418581</td>\n",
       "      <td>rs788023</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>234135973.81</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SF3B1</td>\n",
       "      <td>ENST00000335508</td>\n",
       "      <td>0.658449</td>\n",
       "      <td>0.650674</td>\n",
       "      <td>0.656637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27029</th>\n",
       "      <td>chr2</td>\n",
       "      <td>197393071</td>\n",
       "      <td>rs4685</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>420184758.08</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SF3B1</td>\n",
       "      <td>ENST00000335508</td>\n",
       "      <td>0.658388</td>\n",
       "      <td>0.650572</td>\n",
       "      <td>0.656666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27111</th>\n",
       "      <td>chr2</td>\n",
       "      <td>197400802</td>\n",
       "      <td>rs788018</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>353436214</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SF3B1</td>\n",
       "      <td>ENST00000335508</td>\n",
       "      <td>0.657785</td>\n",
       "      <td>0.650049</td>\n",
       "      <td>0.656028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39318</th>\n",
       "      <td>chr5</td>\n",
       "      <td>112840628</td>\n",
       "      <td>rs42427</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>337788481.33</td>\n",
       "      <td>PASS</td>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>0.652413</td>\n",
       "      <td>0.655320</td>\n",
       "      <td>0.653774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39380</th>\n",
       "      <td>chr5</td>\n",
       "      <td>112840862</td>\n",
       "      <td>rs866006</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>461523359.13</td>\n",
       "      <td>PASS</td>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>0.651259</td>\n",
       "      <td>0.654061</td>\n",
       "      <td>0.652701</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chromosome   Position        rsID REF ALT          QUAL FILTER   Gene  \\\n",
       "21631      chr17   76736425    rs237058   A   G  446471437.84   PASS  SRSF2   \n",
       "17209      chr15   52397434   rs1724577   T   G  686821345.29   PASS  MYO5A   \n",
       "16892      chr15   52375355   rs2414145   G   A  660859405.83   PASS  MYO5A   \n",
       "3400       chr10  121913824  rs10749435   T   C  300183430.26   PASS   ATE1   \n",
       "3158       chr10  121836739   rs4237536   T   C  359903111.56   PASS   ATE1   \n",
       "21695      chr17   76737017    rs237057   G   A  374166392.60   PASS  SRSF2   \n",
       "14484      chr15   41699117   rs2178004   A   T  247403229.11   PASS    MGA   \n",
       "14881      chr15   41734566   rs1918314   C   T  174505923.81   PASS    MGA   \n",
       "39439       chr5  112841059    rs459552   T   A  493715924.54   PASS    APC   \n",
       "15384      chr15   41749514   rs2695163   G   A  357988193.93   PASS    MGA   \n",
       "2744       chr10   43118395   rs1800861   G   T  477129802.32   PASS    RET   \n",
       "2045       chr10   43100520   rs1800858   A   G  419195220.91   PASS    RET   \n",
       "2420       chr10   43111239   rs1800860   A   G  609661534.27   PASS    RET   \n",
       "18485      chr17    7676154   rs1042522   G   C  716976924.97   PASS   TP53   \n",
       "13344      chr12  111446804   rs3184504   T   C  147656611.61   PASS  SH2B3   \n",
       "27363       chr2  197418581    rs788023   T   C  234135973.81   PASS  SF3B1   \n",
       "27029       chr2  197393071      rs4685   T   C  420184758.08   PASS  SF3B1   \n",
       "27111       chr2  197400802    rs788018   A   G     353436214   PASS  SF3B1   \n",
       "39318       chr5  112840628     rs42427   G   A  337788481.33   PASS    APC   \n",
       "39380       chr5  112840862    rs866006   T   G  461523359.13   PASS    APC   \n",
       "\n",
       "            Transcript        AF  controls_AF  non_cancer_AF  \n",
       "21631  ENST00000392485  0.997716     0.997639       0.997741  \n",
       "17209  ENST00000399231  0.968857     0.965944       0.968792  \n",
       "16892  ENST00000399231  0.959533     0.955890       0.959584  \n",
       "3400   ENST00000224652  0.899615     0.894335       0.898484  \n",
       "3158   ENST00000224652  0.895802     0.891226       0.894579  \n",
       "21695  ENST00000392485  0.894510     0.890979       0.894797  \n",
       "14484  ENST00000219905  0.822932     0.822809       0.824158  \n",
       "14881  ENST00000219905  0.800069     0.798792       0.800607  \n",
       "39439  ENST00000257430  0.794825     0.793104       0.794920  \n",
       "15384  ENST00000219905  0.792394     0.790177       0.793146  \n",
       "2744   ENST00000355710  0.739213     0.736804       0.738419  \n",
       "2045   ENST00000355710  0.731156     0.726512       0.730512  \n",
       "2420   ENST00000355710  0.695901     0.694847       0.695013  \n",
       "18485  ENST00000269305  0.668165     0.661152       0.666799  \n",
       "13344  ENST00000341259  0.667391     0.699269       0.673459  \n",
       "27363  ENST00000335508  0.658449     0.650674       0.656637  \n",
       "27029  ENST00000335508  0.658388     0.650572       0.656666  \n",
       "27111  ENST00000335508  0.657785     0.650049       0.656028  \n",
       "39318  ENST00000257430  0.652413     0.655320       0.653774  \n",
       "39380  ENST00000257430  0.651259     0.654061       0.652701  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## 2. Maximum allel frequency each gene\n",
    "ch_variants.sort_values(by = [\"AF\"], ascending=False).head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "tamil-debut",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene\n",
       "SRSF2     0.011968\n",
       "ATE1      0.006188\n",
       "SF3B1     0.005159\n",
       "RET       0.002896\n",
       "ZRSR2     0.002461\n",
       "            ...   \n",
       "MYCN      0.000040\n",
       "MYD88     0.000040\n",
       "KRAS      0.000032\n",
       "SDHAF2    0.000027\n",
       "NRAS      0.000015\n",
       "Name: AF, Length: 65, dtype: float64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## 3. Mean AF per gene\n",
    "ch_variants.groupby([\"Gene\"])[\"AF\"].mean().sort_values(ascending=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pysam",
   "language": "python",
   "name": "pysam"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
